relay-metrics/src/bucket.rs

```diff
@@ -413,10 +413,9 @@ fn parse_tags(string: &str) -> Option<BTreeMap<String, String>> {
             continue;
         }
 
-        let mut value = name_value.next().unwrap_or_default().to_owned();
-        protocol::validate_tag_value(&mut value);
-
-        map.insert(name.to_owned(), value);
+        if let Ok(value) = protocol::unescape_tag_value(name_value.next().unwrap_or_default()) {
```

Comment:
I think we should return `None` if the unescaping fails.

```suggestion
        let value = protocol::unescape_tag_value(name_value.next().unwrap_or_default()).ok()?;
```

Seems to be what we do elsewhere as well.

Reply:
Just a few lines above, we skip individual tags with invalid keys. Are you sure we should break out? That would fail parsing the entire metric.

Reply:
Sorry I missed that, let's keep the behaviour!